numpy/linalg/linalg.py

```diff
@@ -604,6 +606,16 @@ def matrix_power(a, n):
     except TypeError:
         raise TypeError("exponent must be an integer")
 
+    # Fall back on dot for object arrays. Object arrays are not supported by
+    # the current implementation of matmul using einsum
+    if a.dtype != object:
+        fmatmul = matmul
+    elif a.ndim == 2:
```

Comment:
Possibly `<= 2` - what does this do for 1d arrays?

Reply:
1d arrays error. That is tested for.